### PR TITLE
Remove FCHashMapSettings and use FCHashMapConfig instead

### LINKED ISSUE #6377: Remove `FCHashMapSettingsImpl` and move everything to `FcHashConfig`
Refactor every access from `Settings.getFcHashMap()` to use new configuration API

### PR DESCRIPTION
**Pull request by timo0**
Fix #6377 

Removes:
* FCHashMapSettings
* FCHashMapSettingsImpl
* FCHashMapSettingsFactory

### CODE REVIEW

File: platform-sdk/swirlds-fchashmap/src/main/java/com/swirlds/fchashmap/FCHashMap.java [hunk 1]
```diff
@@ -62,6 +62,8 @@
  * 		the type of the value
  */
 public class FCHashMap<K, V> extends AbstractMap<K, V> implements FastCopyable {
+    public static String REBUILD_THREAD_COUNT =
+            String.valueOf(Runtime.getRuntime().availableProcessors());
```
Comment (timo0) [inline platform-sdk/swirlds-fchashmap/src/main/java/com/swirlds/fchashmap/FCHashMap.java:66]: remove

File: platform-sdk/swirlds-fchashmap/src/main/java/com/swirlds/fchashmap/config/FCHashMapConfig.java [hunk 2]
```diff
@@ -29,9 +29,17 @@
...
+ *
  */
 @ConfigData("fcHashMap")
 public record FCHashMapConfig(
         @ConfigProperty(defaultValue = "200") int maximumGCQueueSize,
         @ConfigProperty(defaultValue = "1m") Duration gCQueueThresholdPeriod,
-        @ConfigProperty(defaultValue = "true") boolean archiveEnabled) {}
+        @ConfigProperty(defaultValue = "true") boolean archiveEnabled,
+        @ConfigProperty(defaultValue = "7") int rebuildSplitFactor,
+        // TODO: Runtime.getRuntime().availableProcessors()
```
Comment (timo0) [inline platform-sdk/swirlds-fchashmap/src/main/java/com/swirlds/fchashmap/config/FCHashMapConfig.java:44]: add followup issue to write a ConfigSource

File: platform-sdk/swirlds-fchashmap/src/main/java/com/swirlds/fchashmap/FCHashMap.java (hunk 1 above)
Comment (hendrikebbers) [reply platform-sdk/swirlds-fchashmap/src/main/java/com/swirlds/fchashmap/FCHashMap.java:66]: ?

File: platform-sdk/swirlds-fchashmap/src/main/java/com/swirlds/fchashmap/config/FCHashMapConfig.java [hunk 3]
```diff
@@ -29,9 +29,17 @@
...
+        @ConfigProperty(defaultValue = "4") int rebuildThreadCount) {}
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-fchashmap/src/main/java/com/swirlds/fchashmap/config/FCHashMapConfig.java:45]: We should fix this before the PR can be merged. We can create a custom ConfigSource to calc the value

**Review by hendrikebbers — CHANGES_REQUESTED** [review]

**Review by artemananiev — DISMISSED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/config/sources/ThreadCountPropertyConfigSource.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+package com.swirlds.common.config.sources;
+
+import com.swirlds.config.api.source.ConfigSource;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+public class ThreadCountPropertyConfigSource implements ConfigSource {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/config/sources/ThreadCountPropertyConfigSource.java:25]: Please add Javadoc

File: platform-sdk/swirlds-fchashmap/src/test/java/com/swirlds/fchashmap/FCHashMapTests.java [hunk 5]
```diff
@@ -448,7 +448,7 @@ void insertAndRemoveQuickly() {
     @DisplayName("Null Value Test")
     void nullValueTest() {
         final FCHashMap<Integer, Integer> map = new FCHashMap<>();
-        assertThrows(IllegalArgumentException.class, () -> map.put(0, null), "map should not accept null values");
+        assertThrows(NullPointerException.class, () -> map.put(0, null), "map should not accept null values");
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-fchashmap/src/test/java/com/swirlds/fchashmap/FCHashMapTests.java:451]: why has that changed?

**Review by hendrikebbers — CHANGES_REQUESTED** [review]

File: platform-sdk/swirlds-fchashmap/src/test/java/com/swirlds/fchashmap/FCHashMapTests.java (hunk 5 above)
Comment (timo0) [reply platform-sdk/swirlds-fchashmap/src/test/java/com/swirlds/fchashmap/FCHashMapTests.java:451]: Changed from `CommonUtils.throwArgNull` to `ArgumentUtils.throwArgNull`

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/config/sources/ThreadCountPropertyConfigSource.java (hunk 4 above)
Comment (timo0) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/config/sources/ThreadCountPropertyConfigSource.java:39]: Done

**Review by hendrikebbers — APPROVED** [review]

**Review by alittley — APPROVED** [review]

**Review by imalygin — APPROVED** [review]

**Pull request merged**